<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ExportStateWizard</class>
 <widget class="QWizard" name="ExportStateWizard">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>568</width>
    <height>513</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>300</width>
    <height>320</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Export State</string>
  </property>
  <widget class="QWizardPage" name="wizardPage1">
   <property name="title">
    <string>Export Co-Processing State</string>
   </property>
   <attribute name="pageId">
    <string notr="true">0</string>
   </attribute>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>This wizard will guide you through the steps required to export the current visualization state as a Python script that can be run in the co-processing component of ParaView.  Make sure to add appropriate writers for the desired pipelines to be used in the Writers menu.</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="pqSGExportStateWizardPage2" name="wizardPage2">
   <property name="title">
    <string>Select Simulation Inputs</string>
   </property>
   <property name="subTitle">
    <string>Select the sources in this visualization that are inputs from the simulation. Use Ctrl to select multiple sources.</string>
   </property>
   <attribute name="pageId">
    <string notr="true">1</string>
   </attribute>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0" rowspan="3">
     <widget class="QListWidget" name="allInputs">
      <property name="selectionMode">
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>128</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="2" rowspan="3">
     <widget class="QListWidget" name="simulationInputs">
      <property name="selectionMode">
       <enum>QAbstractItemView::ExtendedSelection</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="addButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Add</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QPushButton" name="removeButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Remove</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="pqSGExportStateWizardPage3" name="wizardPage">
   <property name="title">
    <string>Name Simulation Inputs</string>
   </property>
   <property name="subTitle">
    <string>Assign names for the selected simulation inputs.</string>
   </property>
   <attribute name="pageId">
    <string notr="true">2</string>
   </attribute>
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="QTableWidget" name="nameWidget">
      <property name="alternatingRowColors">
       <bool>true</bool>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SingleSelection</enum>
      </property>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Pipeline Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Simulation Name</string>
       </property>
      </column>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWizardPage" name="wizardPage_2">
   <property name="title">
    <string>Configuration</string>
   </property>
   <property name="subTitle">
    <string>Select state configuration options.</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_4">
    <item row="0" column="0">
     <widget class="QCheckBox" name="liveViz">
      <property name="text">
       <string>Live Visualization</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QSpinBox" name="timeCompartmentSize">
      <property name="toolTip">
       <string>The number of processes working together on a specific time step.</string>
      </property>
      <property name="layoutDirection">
       <enum>Qt::RightToLeft</enum>
      </property>
      <property name="suffix">
       <string/>
      </property>
      <property name="prefix">
       <string/>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Time compartment size</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QCheckBox" name="outputRendering">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>Output rendering components i.e. views</string>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QCheckBox" name="rescaleDataRange">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="toolTip">
       <string>Check to rescale the data range every outputted time step.</string>
      </property>
      <property name="whatsThis">
       <string>Check to rescale the data range every outputted time step.</string>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="text">
       <string>Rescale to Data Range</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0" colspan="2">
     <widget class="QStackedWidget" name="viewsContainer">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>196</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QPushButton" name="previousView">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Previous View</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QPushButton" name="nextView">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Next View</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>pqSGExportStateWizardPage2</class>
   <extends>QWizardPage</extends>
   <header>QWizardPage</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>pqSGExportStateWizardPage3</class>
   <extends>QWizardPage</extends>
   <header>QWizardPage</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>outputRendering</sender>
   <signal>toggled(bool)</signal>
   <receiver>previousView</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>339</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputRendering</sender>
   <signal>toggled(bool)</signal>
   <receiver>nextView</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>272</x>
     <y>339</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputRendering</sender>
   <signal>toggled(bool)</signal>
   <receiver>viewsContainer</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>243</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
